org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -3499,6 +3760,186 @@ class SignatureHelpParams extends TextDocumentPositionParams {
 	}
 }
 
+/**
+ * @since 3.16.0
+ */
+@Beta
+@JsonRpcData
+class SemanticTokensParams {
+
+	/**
+	 * The text document.
+	 */
+	@NonNull
+	TextDocumentIdentifier textDocument
+
+	new(@NonNull TextDocumentIdentifier textDocument) {
+		this.textDocument = Preconditions.checkNotNull(textDocument, 'textDocument')
+	}
+}
+
+/**
+ * @since 3.16.0
+ */
+@Beta
+@JsonRpcData
+class SemanticTokens {
+
+	/**
+	 * An optional result id. If provided and clients support delta updating
+	 * the client will include the result id in the next semantic token request.
+	 * A server can then instead of computing all semantic tokens again simply
+	 * send a delta.
+	 */
+	String resultId
+
+	/**
+	 * The actual tokens.
+	 */
+	@NonNull
+	List<Integer> data
+
+	new(@NonNull List<Integer> data) {
+		this.data = Preconditions.checkNotNull(data, 'data')
+	}
+
+	new(String resultId, @NonNull List<Integer> data) {
+		this(data)
+		this.resultId = resultId
+	}
+}
+
+/**
+ * @since 3.16.0
+ */
+@Beta
+@JsonRpcData
+class SemanticTokensPartialResult {
+
+	@NonNull
+	List<Integer> data
+
+	new(@NonNull List<Integer> data) {
+		this.data = Preconditions.checkNotNull(data, 'data')
+	}
+}
+
+/**
+ * @since 3.16.0
+ */
+@Beta
+@JsonRpcData
+class SemanticTokensDeltaParams {
+
+	/**
+	 * The text document.
+	 */
+	@NonNull
+	TextDocumentIdentifier textDocument
+
+	/**
+	 * The result id of a previous response. The result Id can either point to a full response
+	 * or a delta response depending on what was recevied last.
+	 */
+	@NonNull
+	String previousResultId
+
+	new(@NonNull TextDocumentIdentifier textDocument, @NonNull String previousResultId) {
+		this.textDocument = Preconditions.checkNotNull(textDocument, 'textDocument')
+		this.previousResultId = Preconditions.checkNotNull(previousResultId, 'previousResultId')
+	}
+
+}
+
+/**
+ * @since 3.16.0
+ */
+@Beta
+@JsonRpcData
+class SemanticTokensEdit {
+
+	/**
+	 * The start offset of the edit.
+	 */
+	int start
+
+	/**
+	 * The count of elements to remove.
+	 */
+	int deleteCount;
+
+	/**
+	 * The elements to insert.
+	 */
+	List<Integer> data;
+
+	new(int start, int deleteCount, List<Integer> data) {
+		this.start = start
+		this.deleteCount = deleteCount
+		this.data = data
+	}
+}
+
+/**
+ * @since 3.16.0
+ */
+@Beta
+@JsonRpcData
+class SemanticTokensDelta {
+
+	String resultId
```

Comment:
it's readonly , so it must be in constructor no ?

Reply:
I added a constructor but I think it's accessible via setters too, right?